enterprise/backend/src/metabase_enterprise/semantic_search/index.clj

```diff
@@ -50,6 +50,14 @@
      [[:constraint unique-constraint-name]
       [:unique [:composite :model :model_id]]]]))
 
+(defn sql-format-quoted
+  "Call [[sql/format]] with {:quoted true}.
+
+  Ensures identifiers are quoted since the nano-ids used in temp table names when testing (and various other places)
+  might contain uppercase chars or hyphens which need to be quoted."
```

Comment:
This was the cause of the test flake in the create / drop index tests. Occasionally the temp table names created by `with-temp-index-table!` would contain forbidden chars that postgres converted, and `table-exists-in-db?` would then fail to lookup the table by name in `information_schema`.